File: springmvc/src/main/kotlin/io/github/gaplo917/springblocking/data/BenchmarkData.kt
package io.github.gaplo917.springblocking.data

data class DummyResponse(
  val name: String,
  val isActive: Boolean,
  val isVirtual: Boolean,
  val threadGroupActiveCount: Int,
  val threadGroupCount: Int
) {
  companion object {
    fun dummy(thread: Thread = Thread.currentThread()): DummyResponse {
      return DummyResponse(
        name = thread.name,
        isActive = thread.isAlive,
        isVirtual = thread.isVirtual,
        threadGroupActiveCount = thread.threadGroup.activeCount(),
        threadGroupCount = thread.threadGroup.activeGroupCount()
      )
    }
  }
}

File: frameworks/nestjs/src/data/BenchmarkData.ts
type DummyResponse = {
  name: string | null
  isActive: boolean | null
  isVirtual: boolean | null
  threadGroupActiveCount: number | null
  threadGroupCount: number | null
}

type Thread = {
  name: string,
  isAlive: boolean,
  isVirtual: boolean
  threadGroup: {
    activeCount(): number
    activeGroupCount(): number
  }
}

function dummyResponse(thread?: Thread): DummyResponse {
  return {
    name: thread?.name ?? null,
    isActive: thread?.isAlive ?? null,
    isVirtual: thread?.isVirtual ?? null,
    threadGroupActiveCount: thread?.threadGroup?.activeCount() ?? null,
    threadGroupCount: thread?.threadGroup?.activeGroupCount() ?? null
  }
}

export { dummyResponse, DummyResponse }
